src/app/shared/components/text-input/text-input.component.ts

```diff
@@ -22,6 +22,9 @@ export class TextInputComponent implements ControlValueAccessor {
   @Input() public placeholder?: string
 
   @Input() public helpText: string
+  @Input() public isRequired: boolean
+  @Input() public hasError?: boolean
+  @Input() public errorMessage?: string
```

Comment:
I must admit, I'm very on the fence about these attributes. They are all used for form validation which happens _outside_ the scope of the component, so this component shouldn't know anything about it at all (same applies for all the other components you did this to).

Since validation is part of the form, it should ideally be taken care of by the form.

So instead of baking form validation into components that know nothing about it (because it is outside the scope of the components) we should "build" the form validation when we also build the form.

If we take a look at `EditShelfActionPanelConfigurationComponent`, then right now it looks like:
```
<form ...>
  <sofie-text-input
    ...
    [formControlName]="'name'"
    [isRequired]="true"
    [hasError]="checkControlForError('name')
    errorMessage="..."
  >
  </sofie-text-input>
  <sofie-multi-select
    ...
    [formControlName]="'actionFilter'"
    [isRequired]="true"
    [hasError]="checkControlForError('actionFilter')
    errorMessage="..."
  >
  </sofie-multi-select>
  <sofie-number-input
    ...
    [formControlName]="'rank'"
    [isRequired]="true"
    [hasError]="checkControlForError('rank')
    errorMessage="..."
  >
  </sofie-number-input>
</form>
```
If we "build" the form validation here instead it of giving the validation to components that doesn't know anything about the form, it could look something like this:
```
<form ...>
  <sofie-text-input
    ...
    class="required"
    [formControlName]="'name'"
  >
  </sofie-text-input>
  <sofie-form-error-message *ngIf="checkControlForErrors('name')">Your error message here</sofie-form-error-message>
  <sofie-multi-select
    ...
    class="required"
    [formControlName]="'actionFilter'"
  >
  </sofie-multi-select>
  <sofie-form-error-message *ngIf="checkControlForErrors('actionFiler')">Your error message here</sofie-form-error-message>
  <sofie-number-input
    ...
    class="required"
    [formControlName]="'rank'"
  >
  </sofie-number-input>
  <sofie-form-error-message *ngIf="checkControlForErrors('rank')">Your error message here</sofie-form-error-message>
</form>
```

The above moves the responsibility of form validation away from components that doesn't know what form validation it is. It makes our input components less bloated, because the simply have fewer attributes. And it gives a clearer overview of how the validation of a form works, when looking at the form.

Reply:
@LindvedKrvang That way if we have some margins or/and paddings in to the components as `sofie-text-input`, `sofie-multi-select` the error message will go bellow this spacings, аnd it will make space between end of the component and the error message. In addition I'm not sure I understand what is the idea behind `class="required"`, we want to push the star `(*)` in :after element or am I wrong?

Reply:
The idea behind `class="required"` was more so we didn't need an entire component, `FormRequiredMarkComponent`.

Is the padding/spacing a big issue if the error message isn't baked into our inputs? Ideally, our inputs shouldn't have a lot of padding/margins on themselves, but rather, it's possible to give them padding when building the form.
So you would do something like:
```
<text-input></text-input>
<error-message></error-message>
<text-input class="mt-3"></text-input>
<error-message></error-message>
```